### PR TITLE
Add type_as attribute to specta

### PR DESCRIPTION
**Pull request by mablin7**
We were looking into using specta in our project, but ran into the issue of needing an escape hatch for certain types that can't (easily) implement `Type`.  This solution is similar to ts_rs's `#[ts(type = "..")]` attribute.

`#[specta(type_as = <Ident>)]` on a field overrides its type in the generated type definition with the type of the identifier.

Example:
```rust
use specta::{ts::ts_export, Type};

pub enum MyEnum {
    A,
    B,
    C,
}

#[derive(Type)]
pub struct MyCustomType {
    #[specta(type_as=String)]
    pub nested: MyEnum,
}

fn main() {
    println!("{}", ts_export::<MyCustomType>().unwrap(),);
}
```

Any feedback is super appreciated!

### CODE REVIEW

File: crates/specta/tests/ts_rs/field_override.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+#[derive(Type)]
+struct Override {
+    a: i32,
+    #[specta(type_as = String)]
+    b: i32,
+}
+
+#[test]
+fn test() {
```
Comment (Brendonovich) [inline crates/specta/tests/ts_rs/field_override.rs:11]: The `ts_rs` tests are meant to be close reflections of `ts-rs`'s actual tests - the relevant one for type overrides is [here](https://github.com/Aleph-Alpha/ts-rs/blob/f2dbbc8c10ba7071acddbe69480edfd7cfe0fbc7/ts-rs/tests/type_override.rs). Just change the file name and make the `ts-rs` test work with Specta and it's fine.

File: crates/specta/examples/type_as.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    C,
+}
+
+#[derive(Type)]
+pub struct MyCustomType {
+    #[specta(type_as=String)]
+    pub nested: MyEnum,
+}
+
+fn main() {
```
Comment (Brendonovich) [inline crates/specta/examples/type_as.rs:15]: Idk if an entire example is needed but ehh I'm nitpicking

File: crates/specta/macros/src/type/attr/field.rs [hunk 3]
```diff
@@ -1,10 +1,11 @@
-use syn::{Attribute, Result};
+use syn::{Attribute, Result, Token, Type};
 
 use crate::utils::*;
 
 #[derive(Default)]
 pub struct FieldAttr {
     pub rename: Option<String>,
+    pub type_as: Option<Type>,
```
Comment (Brendonovich) [inline crates/specta/macros/src/type/attr/field.rs:8]: `ts-rs` uses the `type` option, which I think is a nicer name than `type_as`

**Review by Brendonovich — CHANGES_REQUESTED** [review]
Thanks for working on this! Super cool idea to override the Rust type that is used rather than writing custom typescript.
I'd like the feature to mirror `ts-rs` a bit more, but you've already done quite a good job.
I might rebase this PR onto a bunch of new changes to Specta that are being made in preparation to make it more of a standalone library, I'll handle that though.

**Comment by Brendonovich** [comment]
Also would you mind explaining why you need type overrides, and why you picked Specta over `ts-rs`? Just curious haha

**Review by Brendonovich — APPROVED** [review]

**Review by oscartbeaumont — APPROVED** [review]
Thanks for your time. Looks great!

**Comment by mablin7** [comment]
> Also would you mind explaining why you need type overrides, and why you picked Specta over `ts-rs`? Just curious haha

Oh, sorry I didn't have the time to address your comments, thanks for the corrections.

Our main problem was that the messages in our tauri app included types from 3rd party crates that we normally just serialize to string, but we can't have them implement Type. 

The reason we would like to use specta is its ability to generate all referenced type definitions recursively. With ts-rs we have a huge list of types that need to be printed, and it needs to be updated manually. Whereas with specta, it would just be a single call, since we already have all relevant types in an enum anyway.

**Pull request merged**